logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/e2ei/usecase/GetUserE2EIAllCertificatesUseCase.kt

```diff
@@ -27,23 +27,28 @@ import com.wire.kalium.logic.data.conversation.ClientId
 
 /**
  * This use case is used to get all e2ei certificates of the user.
- * Returns Map<String, E2eiCertificate> where key is value of [ClientId] and [E2eiCertificate] is certificate itself
+ * Returns [UsersE2eiCertificates] from which we can get [E2eiCertificate] by [ClientId]
  */
 interface GetUserE2eiCertificatesUseCase {
-    suspend operator fun invoke(userId: UserId): Map<String, E2eiCertificate>
+    suspend operator fun invoke(userId: UserId): UsersE2eiCertificates
 }
 
 class GetUserE2eiCertificatesUseCaseImpl internal constructor(
     private val mlsConversationRepository: MLSConversationRepository,
     private val pemCertificateDecoder: PemCertificateDecoder
 ) : GetUserE2eiCertificatesUseCase {
-    override suspend operator fun invoke(userId: UserId): Map<String, E2eiCertificate> =
+    override suspend operator fun invoke(userId: UserId): UsersE2eiCertificates =
         mlsConversationRepository.getUserIdentity(userId).map { identities ->
             val result = mutableMapOf<String, E2eiCertificate>()
             identities.forEach {
                 result[it.clientId] = pemCertificateDecoder.decode(it.certificate, it.status)
             }
-            result
-        }.getOrElse(mapOf())
+            UsersE2eiCertificates(userId, result)
+        }.getOrElse(UsersE2eiCertificates(userId, mapOf()))
 
 }
+
+data class UsersE2eiCertificates(private val userId: UserId, private val map: Map<String, E2eiCertificate>) {
+    operator fun get(clientId: ClientId): E2eiCertificate? = map["${userId.value}:${clientId.value}@${userId.domain}"]
```

Comment:
question: is this `"${userId.value}:${clientId.value}@${userId.domain}"` the same as `QualifiedClientID`. If that's the case makes sense to return it here?

Reply:
well, it's not the same, this ClientId is just a `String` which is kinda combination of `UserId` and `ClientId` values. 
So I could pars it into `QualifiedClientID` but i don't think it makes much sense.